server/store/memory/peer_store_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package memory
+
+import (
+	"net"
+	"testing"
+
+	"github.com/chihaya/chihaya"
+	"github.com/chihaya/chihaya/server/store"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	hash1 = chihaya.InfoHash("11111111111111111111")
+	hash2 = chihaya.InfoHash("22222222222222222222")
+	hash3 = chihaya.InfoHash("33333333333333333333")
+
+	peers = []chihaya.Peer{
+		chihaya.Peer{chihaya.PeerID("uPk9xHM3mLLoGFQ9zf6L"), net.ParseIP("250.183.81.177"), 5720},
+		chihaya.Peer{chihaya.PeerID("q4iZnzlj1ixvwjtnzOB7"), net.ParseIP("231.231.49.173"), 1453},
+		chihaya.Peer{chihaya.PeerID("vfEXFmVQ5tQklXIcoNnt"), net.ParseIP("fd45:7856:3dae::48"), 2878},
+		chihaya.Peer{chihaya.PeerID("Q0As1KlIMfwesEiT0PJn"), net.ParseIP("fdad:c435:bf79::12"), 4114},
+		chihaya.Peer{chihaya.PeerID("ulJDilO3euW0AjoFgYAe"), net.ParseIP("254.99.84.77"), 7032},
+		chihaya.Peer{chihaya.PeerID("75PYue4EcP9ruBopFLJE"), net.ParseIP("fd0a:29a8:8445::38"), 3167},
+		chihaya.Peer{chihaya.PeerID("X4gYxRI9llLq1P3ugwTv"), net.ParseIP("fdda:1b35:7d6e::9"), 6179},
+		chihaya.Peer{chihaya.PeerID("sB1yBRic6KLZFCo2HHxq"), net.ParseIP("38.241.13.19"), 4833},
+		chihaya.Peer{chihaya.PeerID("rjChdTACqLDWnSYGaxnq"), net.ParseIP("211.229.100.129"), 2614},
+		chihaya.Peer{chihaya.PeerID("zaN96RpTId8At8X8yBUG"), net.ParseIP("fd7f:78f0:4c77::55"), 4727},
+	}
+)
+
+func peerInSlice(peer chihaya.Peer, peers []chihaya.Peer) bool {
+	for _, v := range peers {
+		if v.ID == peer.ID && v.Port == peer.Port {
+			return true
+		}
+	}
+	return false
+}
+
+func TestPeerStore(t *testing.T) {
+	var d = &peerStoreDriver{}
+	var err error
+	var ret bool
+	var num uint32
+
+	s, err := d.New(&store.DriverConfig{
+		"memory",
+		struct{ Shards int }{1},
+	})
+	assert.Nil(t, err)
+	assert.NotNil(t, s)
+
+	// add all peers, odd indexes are seeders, and even are leechers
```

Comment:
This is very error-prone. You should encode this into your test data as in my previous comment.